Doctrine/Phpcr/SitemapUrlInformationProvider.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2014 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\SeoBundle\Doctrine\Phpcr;
+
+use Doctrine\ODM\PHPCR\DocumentManager;
+use Jackalope\Transport\Logging\LoggerInterface;
+use PHPCR\Query\QueryInterface;
+use Psr\Log\LoggerInterface as PsrLogger;
+use Symfony\Cmf\Bundle\CoreBundle\PublishWorkflow\PublishWorkflowChecker;
+use Symfony\Cmf\Bundle\SeoBundle\AlternateLocaleProviderInterface;
+use Symfony\Cmf\Bundle\SeoBundle\Extractor\ExtractorInterface;
+use Symfony\Cmf\Bundle\SeoBundle\Model\UrlInformation;
+use Symfony\Cmf\Bundle\SeoBundle\SeoPresentation;
+use Symfony\Cmf\Bundle\SeoBundle\Sitemap\UrlInformationProviderInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\SecurityContextInterface;
+
+/**
+ * Providers are able to create a list of UrlInformation for the sitemap creation.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@gmx.de>
+ */
+class SitemapUrlInformationProvider implements UrlInformationProviderInterface
+{
+    /**
+     * @var AlternateLocaleProviderInterface
+     */
+    protected $alternateLocaleProvider;
+
+    /**
+     * @var ExtractorInterface
+     */
+    protected $titleExtractor;
+
+    /**
+     * @var DocumentManager
+     */
+    private $manager;
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var SecurityContextInterface
+     */
+    private $publishWorkflowChecker;
+    /**
+     * @var string
+     */
+    private $defaultChanFrequency;
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+    /**
+     * @var SeoPresentation
+     */
+    private $seoPresentation;
+
+    /**
+     * @param DocumentManager $manager
+     * @param RouterInterface $router
+     * @param string $defaultChanFrequency
+     * @param PsrLogger $logger
+     * @param SeoPresentation $seoPresentation
+     * @param SecurityContextInterface $publishWorkflowChecker
+     */
+    public function __construct(
+        DocumentManager $manager,
+        RouterInterface $router,
+        $defaultChanFrequency,
+        PsrLogger $logger,
+        SeoPresentation $seoPresentation,
+        SecurityContextInterface $publishWorkflowChecker = null
+    ) {
+        $this->manager = $manager;
+        $this->router = $router;
+        $this->defaultChanFrequency = $defaultChanFrequency;
+        $this->logger = $logger;
+        $this->seoPresentation = $seoPresentation;
+        $this->publishWorkflowChecker = $publishWorkflowChecker;
+    }
+
+    /**
+     * {@inheritDocs}
+     */
+    public function getUrlInformation()
+    {
+        $contentDocuments = $this->fetchSitemapDocuments();
+
+        return $this->computeUrlInformationFromSitemapDocuments($contentDocuments);
```

Comment:
i would suggest to keep the loop and published check in this method, and have the compute only handle a single document per call. makes it easier if i want to customize the compute